simulator/mips/mips_instr.h

```diff
@@ -24,6 +24,38 @@
 #include <array>
 #include <unordered_map>
 
+enum instr_masks {
+    //32bits         01234567012345670123456701234567
+    FUNCT_MASK    = 0b00000000000000000000000000111111,
+    SHAMT_MASK    = 0b00000000000000000000011111000000,
+    RD_MASK       = 0b00000000000000001111100000000000,
+    RT_MASK       = 0b00000000000111110000000000000000,
+    RS_MASK       = 0b00000011111000000000000000000000,
+    OPCODE_MASK   = 0b11111100000000000000000000000000,
+    ASI_IMM_MASK  = 0b00000000000000001111111111111111,
+    ASJ_IMM_MASK  = 0b00000011111111111111111111111111,
+};
+
+enum instr_size_and_shifts {
+    FUNCT_SIZE    = 6,
```

Comment:
We may use our `bitcount` constexpr macro (infra/macro.h) instead of manual variables

Reply:
Oops, sorry, we do not have one. I'll implement it in a minute...

Reply:
Ah, it is called `popcount`...